vote/src/vote_account.rs

```diff
@@ -34,17 +38,17 @@ pub enum Error {
 #[derive(Debug)]
 struct VoteAccountInner {
     account: AccountSharedData,
-    vote_state: OnceLock<Result<VoteState, Error>>,
+    vote_state: VoteState,
 }
 
 pub type VoteAccountsHashMap = HashMap<Pubkey, (/*stake:*/ u64, VoteAccount)>;
-
 #[cfg_attr(feature = "frozen-abi", derive(AbiExample))]
-#[derive(Clone, Debug, Deserialize)]
-#[serde(from = "Arc<VoteAccountsHashMap>")]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct VoteAccounts {
+    #[serde(deserialize_with = "deserialize_accounts_hash_map")]
     vote_accounts: Arc<VoteAccountsHashMap>,
     // Inner Arc is meant to implement copy-on-write semantics.
+    #[serde(skip)]
```

Comment:
Was this needed? We weren't serializing the inner map before were we?

Reply:
We had a custom Serialize impl to skip this field. We had a custom Deserialize (from = ...) that skipped it too. I've removed the custom deserialize because now we need custom deserialize for `vote_accounts` so the option is to either add a new `impl Deserialize for VoteAccounts` that skips... or just add this attribute (and remove the custom Serialize since it can be achieved with the attribute too).

Reply:
Ah right, I see that now, thanks looks fine. Looks like we can now remove the derived `Deserialize` and `serde(try_from)` from `VoteAccount.

Reply:
Correct! I didn't because then I have to remove two tests and the diff gets bigger :P But happy to do it if you want!

Reply:
Yeah go for it. I think it's confusing to leave it in there after the changes in this PR

Reply:
done 1978c05